src/realm/sync/noinst/client_history_impl.cpp

```diff
@@ -376,133 +376,161 @@ void ClientHistory::find_uploadable_changesets(UploadCursor& upload_progress, ve
 }
 
 
-void ClientHistory::integrate_server_changesets(const SyncProgress& progress,
-                                                const std::uint_fast64_t* downloadable_bytes,
-                                                util::Span<const RemoteChangeset> incoming_changesets,
-                                                VersionInfo& version_info, DownloadBatchState batch_state,
-                                                util::Logger& logger,
-                                                util::UniqueFunction<void(const TransactionRef&)> run_in_write_tr,
-                                                SyncTransactReporter* transact_reporter)
+void ClientHistory::integrate_server_changesets(
+    const SyncProgress& progress, const std::uint_fast64_t* downloadable_bytes,
+    util::Span<const RemoteChangeset> incoming_changesets, VersionInfo& version_info, DownloadBatchState batch_state,
+    util::Logger& logger, util::UniqueFunction<void(const TransactionRef&, size_t)> run_in_write_tr,
+    SyncTransactReporter* transact_reporter, bool is_flx_sync_connection)
```

Comment:
I always find adding a boolean flag to a big complicated function is a bit of a code smell because you can't see what `true` means by looking at the callsite. I have a suggestion below that might make this clearer though.